index.ts

```diff
@@ -822,6 +876,51 @@ class IntegrationAPI extends EventEmitter {
   public updateEntityAttributes(entityId: string, attributes: { [key: string]: string | number | boolean }): boolean {
     return this.#configuredEntities.updateEntityAttributes(entityId, attributes);
   }
+
+  public async generateOauth2AuthUrl(): Promise<{ auth_url: string }> {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("generateOauth2AuthUrl: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.GenerateOauth2AuthUrl);
+  }
```

Comment:
Parameter missing for the optional `client_data` dictionary:
- https://github.com/unfoldedcircle/core-api/blob/77f3eed73a1c0e29e9aeacfa3026f5a71e1ebf21/integration-api/UCR-integration-asyncapi.yaml#L773-L775
- https://github.com/unfoldedcircle/core-api/blob/77f3eed73a1c0e29e9aeacfa3026f5a71e1ebf21/integration-api/UCR-integration-asyncapi.yaml#L1480-L1502